pkg/operators/kubenetworkmanager/kubenetworkmanager.go

```diff
@@ -0,0 +1,322 @@
+type k8sInventoryCache struct {
+	clientset *kubernetes.Clientset
+
+	pods atomic.Pointer[v1.PodList]
+	svcs atomic.Pointer[v1.ServiceList]
+
+	exit   chan bool
+	ticker *time.Ticker
+}
+
+func newCache(tickerDuration time.Duration) (*k8sInventoryCache, error) {
+	clientset, err := k8sutil.NewClientset("")
+	if err != nil {
+		return nil, fmt.Errorf("creating new k8s clientset: %w", err)
+	}
+
+	cache := &k8sInventoryCache{
+		clientset: clientset,
+		exit:      make(chan bool),
+		ticker:    time.NewTicker(tickerDuration),
+	}
+
+	// Store empty lists
+	cache.pods.Store(&v1.PodList{})
+	cache.svcs.Store(&v1.ServiceList{})
+
+	go cache.loop()
+
+	return cache, nil
+}
+
+func (cache *k8sInventoryCache) loop() {
+	for {
+		select {
+		case <-cache.exit:
+			return
+		case <-cache.ticker.C:
+			cache.update()
+		}
+	}
+}
+
+func (cache *k8sInventoryCache) Close() {
+	cache.exit <- true
+	cache.ticker.Stop()
+}
+
+func (cache *k8sInventoryCache) update() {
+	pods, err := cache.clientset.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("listing pods: %v", err)
+		return
+	}
+	cache.pods.Store(pods)
+
+	svcs, err := cache.clientset.CoreV1().Services("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Errorf("listing services: %s", err)
+		return
+	}
+	cache.svcs.Store(svcs)
+}
+
+func (cache *k8sInventoryCache) GetPods() *v1.PodList {
+	return cache.pods.Load()
+}
+
+func (cache *k8sInventoryCache) GetSvcs() *v1.ServiceList {
+	return cache.svcs.Load()
+}
+
+type KubeNetworkManager struct {
+	k8sInventory *k8sInventoryCache
+}
+
+func (k *KubeNetworkManager) Name() string {
+	return "KubeNetworkManager"
+}
+
+func (k *KubeNetworkManager) Description() string {
+	return "KubeNetworkManager handles more descriptive informations about remote ips"
+}
+
+func (k *KubeNetworkManager) GlobalParamDescs() params.ParamDescs {
+	return nil
+}
+
+func (k *KubeNetworkManager) ParamDescs() params.ParamDescs {
+	// TODO: Write patch so empty paramsDescs is okay
+	return params.ParamDescs{
+		{
+			Key:         "asdfsadf",
+			Description: "panics without at least one param",
+		},
+	}
+}
+
+func (k *KubeNetworkManager) Dependencies() []string {
+	return []string{"KubeManager"}
+}
+
+func (k *KubeNetworkManager) CanOperateOn(gadget gadgets.GadgetDesc) bool {
+	km := kubemanager.KubeManager{}
+	if !km.CanOperateOn(gadget) {
+		return false
+	}
+
+	gi, ok := gadget.(gadgets.GadgetInstantiate)
+	if !ok {
+		return false
+	}
+
+	instance, err := gi.NewInstance()
+	if err != nil {
+		log.Printf("failed to create dummy instance")
+		return false
+	}
+
+	_, hasResolver := instance.(SetIPToK8sResolverInterface)
+	_, hasAdditionalInf := gadget.EventPrototype().(AdditionalPodNetworkInformation)
+
+	return hasResolver && hasAdditionalInf
+}
+
+func (k *KubeNetworkManager) Init(params *params.Params) error {
+	k8sInventory, err := newCache(1 * time.Second)
```

Comment:
IMHO this should happen "on demand" for now. We discussed to maybe add some intelligence to operators later on, that make sure operators are unloaded when they're not used. But we're not there, yet.
Especially in this case it means that right now once this operator has been used for a trace, it will keep updating the cache every second, which seems unnecessary stress to me. So maybe we should add and maintain a reference counter in PreGadgetRun/PostGadgetRun to stop the cache ticker once that counter hits 0 again.

Reply:
Implemented such a thing with a `mutex`. Probably could be done with `atomic.Integer.Add()` or so. But the `PreGadgetRun`/`PostGadgetRun` wont be called that much anyway.